torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -578,6 +579,27 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processMessage(
          rpc = (std::shared_ptr<RpcCommandBase>)std::move(rpc),
          messageType = request.type(),
          id = request.id()]() {
+          // This cost of pre-request check is minimal thanks to
+          // std::shared_lock. The cost is in magnitude
+          // of 10us. If server global profiler is enabled, we futher pay the
+          // cost of thread local profiler state initialization .
+          // int64_t startTime = ::torch::autograd::profiler::getTime();
+          bool isServerProcessGlobalProfilerEnabled =
+              serverProcessGlobalProfilerEnabled();
+          // int64_t endTime = ::torch::autograd::profiler::getTime();
+          // LOG(ERROR) << "Test profiler enabled took: "
+          //            << (endTime - startTime) / 1000 << " us";
+          std::shared_ptr<ProcessGlobalProfilerState>
+              serverProcessGlobalProfilerStatePtr;
+          if (isServerProcessGlobalProfilerEnabled) {
+            // Initialize thread-local profiler state from process-global
+            // profiler state.
+            serverProcessGlobalProfilerStatePtr =
```

Comment:
This is in the RRef ready callback, which can run on a different thread. Is this intentional?

Reply:
Yes, intentional. It doesn't matter.